packages/xo-web/src/common/xo/index.js

```diff
@@ -2869,7 +2869,16 @@ import NewSshKeyModalBody from './new-ssh-key-modal' // eslint-disable-line impo
 export const addSshKey = key => {
   const { preferences } = xo.user
   const otherKeys = (preferences && preferences.sshKeys) || []
+  const sshKeyAlreadyExistsWarning = ({ key }) => {
+    if (otherKeys.some(otherKey => otherKey.key === key)) {
+      error(_('sshKeyErrorTitle'), _('SshKeyAlreadyExists'))
+      return
+    }
+  }
+
   if (key) {
+    sshKeyAlreadyExistsWarning(key)
```

Comment:
Should we set the user preference if the key already exists?

Reply:
Why? if it already exists, we don't need to save it and that's why we have this issue.

Reply:
With this implementation, the key will be always saved even if it exists.